crates/common/src/paths.rs

```diff
@@ -36,6 +53,41 @@ pub fn resolve_manifest_file_path(provided_path: impl AsRef<Path>) -> Result<Pat
     }
 }
 
+/// Starting from the current directory, searches upward through
+/// the directory tree for a manifest (that is, a file with the default
+/// manifest name `spin.toml`). If found, the path to the manifest
+/// is returned, with a boolean flag indicating if the found path was
+/// the default (i.e. `spin.toml` in the current directory).
+/// If no matching file is found, the function returns None.
+///
+/// The search is abandoned if it reaches the root directory, or the
+/// root of a Git repository, without finding a 'spin.toml'.
+pub fn search_upwards_for_manifest() -> Option<(PathBuf, usize)> {
+    let candidate = PathBuf::from(DEFAULT_MANIFEST_FILE);
+
+    if candidate.is_file() {
+        return Some((candidate, 0));
+    }
+
+    for distance in 1..20 {
```

Comment:
20 seems a little bit high but no strong opinions. Maybe this is something we want to be configurable to users using env vars?

Reply:
Configuring it seems wildly excessive even by the standards of great-great-great-grandparent string code.  This was randomly picked to be high enough that nobody will ever run into it but low enough that if we hit some kind of crazy infinite loop case then it will exit before taking too long.  But happy to reduce it.  I originally had it at 10.  What am I bid?

(Price Is Right audience "higher! lower!" sound FX)

Reply:
@karthik2804 I'm going to punt on this since it's non-blocking - happy to tweak as a follow-up.